homeassistant/config_entries.py

```diff
@@ -2103,32 +2075,32 @@ async def async_forward_entry_setup(
         Instead, await async_forward_entry_setups as it can load
         multiple platforms at once and is more efficient since it
         does not require a separate import executor job for each platform.
-
-        If platforms must be loaded late (after the config entry is setup),
-        use async_late_forward_entry_setup instead.
         """
-        if non_locked_platform_forwards := not entry.setup_lock.locked():
-            _report_non_locked_platform_forwards(entry)
-        else:
-            report(
-                "calls async_forward_entry_setup for "
-                f"integration, {entry.domain} with title: {entry.title} "
-                f"and entry_id: {entry.entry_id}, which is deprecated and "
-                "will stop working in Home Assistant 2025.6, "
-                "await async_forward_entry_setups instead",
-                error_if_core=False,
-                error_if_integration=False,
-            )
-        return await self._async_forward_entry_setup(
-            entry, domain, True, non_locked_platform_forwards
+        report(
+            "calls async_forward_entry_setup for "
+            f"integration, {entry.domain} with title: {entry.title} "
+            f"and entry_id: {entry.entry_id}, which is deprecated and "
+            "will stop working in Home Assistant 2025.6, "
+            "await async_forward_entry_setups instead",
+            error_if_core=False,
+            error_if_integration=False,
         )
+        if not entry.setup_lock.locked():
+            async with entry.setup_lock:
+                if entry.state is not ConfigEntryState.LOADED:
+                    raise OperationNotAllowed(
+                        f"The config entry {entry.title} ({entry.domain}) with entry_id"
+                        f" {entry.entry_id} cannot forward setup for {domain} because it"
+                        f" is not loaded in the {entry.state} state"
+                    )
+                return await self._async_forward_entry_setup(entry, domain, True)
```

Comment:
This is a bit more WET than I would have liked but since this function will go away anyways its not a concern long term